### PR TITLE
TST: collect arithmetic test helpers

### PR DESCRIPTION
**Pull request by jbrockmendel**
broken off from a WIP branch that parametrizes/fixturizes/de-duplicates a bunch of timedelta64 arithmetic tests

### CODE REVIEW

File: pandas/tests/arithmetic/test_datetime64.py [hunk 1]
```diff
@@ -31,54 +31,11 @@
...
-    with pytest.raises(TypeError, match=msg):
-        right < left
-    with pytest.raises(TypeError, match=msg):
-        right <= left
-    with pytest.raises(TypeError, match=msg):
-        right > left
-    with pytest.raises(TypeError, match=msg):
-        right >= left
-
+from .common import (
```
Comment (jreback) [inline pandas/tests/arithmetic/test_datetime64.py]: are we using relative imports anywhere?

Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_datetime64.py]: changed to absolute

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**